packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -2091,10 +2091,16 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
     if (widget.obscureText) {
       String text = _value.text;
       text = widget.obscuringCharacter * text.length;
-      final int o =
-        _obscureShowCharTicksPending > 0 ? _obscureLatestCharIndex : null;
-      if (o != null && o >= 0 && o < text.length)
-        text = text.replaceRange(o, o + 1, _value.text.substring(o, o + 1));
+      // Obscure latest character only on mobile
+      if ((defaultTargetPlatform == TargetPlatform.android ||
+              defaultTargetPlatform == TargetPlatform.iOS ||
+              defaultTargetPlatform == TargetPlatform.fuchsia) &&
+          !kIsWeb) {
```

Comment:
It sounds like we should make sure that we get these platforms correct.

@yjbanov You [commented](https://github.com/flutter/flutter/issues/51218#issuecomment-592230646) about this not being an issue on web.  Do you want the behavior to stay as-is on web?  Or @goderbauer mentioned that Safari on iOS shows the last character, do we want to try to match that specific behavior?